src/proxy.rs

```diff
@@ -732,19 +729,18 @@ fn get_sys_proxies(
         RegistryProxyValues,
     >,
 ) -> SystemProxyMap {
-    let proxies = get_from_environment();
-
     // TODO: move the following #[cfg] to `if expression` when attributes on `if` expressions allowed
     #[cfg(target_os = "windows")]
     {
+        let proxies = get_from_environment();
         if proxies.is_empty() {
             // don't care errors if can't get proxies from registry, just return an empty HashMap.
             if let Some(registry_values) = registry_values {
                 return parse_registry_values(registry_values);
             }
         }
     }
-    proxies
+    get_from_environment()
```

Comment:
I think this change would make it so that on Windows, the environment would be checked twice. I think it's also clearer the way it was previously.

Reply:
@seanmonstar, Thanks for your suggestion, but I guess as per the current code the validation for checking it twice is commented and marked for future releases, and for now we are just creating an extra variable. But still, if you want I can remove this change as well.

Thanks